### DISCUSSION TITLE
How to make RC5 encodings?

### DISCUSSION THREAD

**Discussion by jnpwly** [arendst/Tasmota#15676] [Q&A] [answered]
Hello Tasmota-ians 👋

I am trying to figure out how to send some RC5 encoded IR signals from my IR "puck"... but am stumped when trying to work out how to specify the data to send. Is anyone able to give me a hand?

There's a really useful spreadsheet that Marantz have made available, here: https://www.marantz.com/-/media/files/documentmaster/marantzna/us/marantz-2014-ir-command-sheet.xls. From this, it looks like I should be able turn the Receiver on or off by sending a "remote code" of `16 12` (System: 16, Command: 12, Extension: none). However, when I capture the IR signals from the remote control when pressing the power button, I see:

```
05:22:18.933 RSL: RESULT = {"IrReceived":{"Protocol":"RC5","Bits":12,"Data":"0x40C","DataLSB":"0x2030","Repeat":0}}
05:22:19.083 RSL: RESULT = {"IrReceived":{"Protocol":"RC5","Bits":12,"Data":"0x40C","DataLSB":"0x2030","Repeat":0}}
05:22:19.185 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":9,"Hash":"0x53513047","Repeat":0}}
05:22:19.283 RSL: RESULT = {"IrReceived":{"Protocol":"RC5","Bits":12,"Data":"0x40C","DataLSB":"0x2030","Repeat":0}}
05:22:19.433 RSL: RESULT = {"IrReceived":{"Protocol":"RC5","Bits":12,"Data":"0x40C","DataLSB":"0x2030","Repeat":0}}
05:22:19.535 RSL: RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":8,"Hash":"0x2108C4E6","Repeat":0}}
```

Obviously there's some repitition going on there... but I don't understand how `"Data":"0x40C","DataLSB":"0x2030"` or `"Data":"0x40C","DataLSB":"0x2030"` might correspond to RC5 "codes" -- specifically, a "remote code" of 16 12 -- given in the spreadsheet.

I can, by trial and error, capture the IR signals from each button on the remote... but it looks like there are a number of additional "commands" available which are not available from the remote -- for example, setting the volume to a specific value: the spreadsheet indicates a "remote code" of `16 111 31` will set the volume to 50%.

So, is anyone able to help me understand how to convert the details given in that spreadsheet into a Tasmota command?

Many thanks,
   Jon

**Comment by sfromis** [comment, marked as answer]
Did you read my answer on Discord? To repeat:

> While I know no details, the single example you posted does have the similarity that "Data":"0x40C"  ends with 0C which is hexadecimal for decimal 12, the command code you mentioned.

**Reply by jnpwly** [reply]
Thanks, @sfromis. 

"While I know no details..." I think I'm precisely looking for details, as I've exhausted my experimentation and trial-and-error so far.

**Reply by sfromis** [reply]
I posted one very specific observation which very much appears to be relevant. Does that match other codes?
